pkgs/applications/misc/double-commander/default.nix

```diff
@@ -0,0 +1,104 @@
+{
+  stdenv, lib
+  , fetchFromGitHub
+  , fpc, lazarus
+  , which, autoPatchelfHook
+  , libqt5pas, qt5, gtk2, dbus
+}:
+
+let qtLazarus = lazarus.override {
+  withQt = true;
+  inherit libqt5pas;
+};
+in
+stdenv.mkDerivation rec {
+  pname = "doublecmd";
+  version = "1.0.2";
+
+  src = fetchFromGitHub {
+    owner = "doublecmd";
+    repo = "doublecmd";
+    rev = "v${version}";
+    sha256 = "sha256-nX5EZekYF9saM0QaN9caCnsRDIqPYNX/BNu6Bb5lDYQ=";
+  };
+
+  nativeBuildInputs = [
+    fpc
+    which
+    autoPatchelfHook
+    qt5.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    qt5.qtbase
+    libqt5pas
+    gtk2
+    dbus.lib
+  ];
+
+  # dontWrapQtApps = true;
+
+  buildPhase = ''
+    export HOME=$(mktemp -d)
+    LAZ_FLAGS="--widgetset=qt5 --cpu=x86_64"
+    lazbuild="${qtLazarus}/bin/lazbuild --lazarusdir=${qtLazarus}/share/lazarus"
+
+    # build components
+    $lazbuild components/CmdLine/cmdbox.lpk
+    $lazbuild components/doublecmd/doublecmd_common.lpk
+    $lazbuild components/KASToolBar/kascomp.lpk
+    $lazbuild components/chsdet/chsdet.lpk
+    $lazbuild components/dcpcrypt/dcpcrypt.lpk
+    $lazbuild components/gifanim/pkg_gifanim.lpk
+    # $lazbuild components/lclextensions/lclextensions_package_doublecmd.lpk  # doesn't seem to be necessary?
+    $lazbuild components/multithreadprocs/multithreadprocslaz.lpk
+    $lazbuild components/synunihighlighter/synuni.lpk
+    $lazbuild components/viewer/viewerpackage.lpk
+    # $lazbuild components/virtualtreeview/virtualtreeview_package_doublecmd.lpk  # doesn't seem to be necessary?
+
+    # this will replace cmdbox in the next release
+    # $lazbuild components/virtualterminal/virtualterminal.lpk
```

Comment:
```suggestion
    lazbuild --lazarusdir=${lazarus}/share/lazarus --pcp=./lazarus --ws=${widgetset} \
      components/chsdet/chsdet.lpk \
      components/CmdLine/cmdbox.lpk \
      components/multithreadprocs/multithreadprocslaz.lpk \
      components/dcpcrypt/dcpcrypt.lpk \
      components/doublecmd/doublecmd_common.lpk \
      components/KASToolBar/kascomp.lpk \
      components/viewer/viewerpackage.lpk \
      components/gifanim/pkg_gifanim.lpk \
      components/synunihighlighter/synuni.lpk \
      src/doublecmd.lpi
```

Reply:
applied in fe0fe8d6b0ebe88ed7bff6309c3b786592a9d2bd